### PR TITLE
Add setImplementation for TUSD

### PR DESCRIPTION
**Pull request by andy8052**

### CODE REVIEW

File: src/dss/GemJoinImplementationAbstract.sol [hunk 1]
```diff
@@ -0,0 +1,17 @@
+pragma solidity >=0.5.12;
+
+// https://github.com/makerdao/dss/blob/master/src/join.sol
```
Comment (lucas-manuel) [inline src/dss/GemJoinImplementationAbstract.sol:3]: LGTM, I would just change this to: https://github.com/makerdao/dss-deploy/blob/master/src/join.sol

**Review by lucas-manuel — APPROVED** [review]

File: src/dss/GemJoinImplementationAbstract.sol (hunk 1 above)
Comment (andy8052) [reply src/dss/GemJoinImplementationAbstract.sol]: ah yeah good catch

**Comment by andy8052** [comment]
I am going to merge since this is low impact and I need it for the spell. If we have a retroactive fix we can do that still

File: src/dss/GemJoinImplementationAbstract.sol (hunk 1 above)
Comment (brianmcmichael) [reply src/dss/GemJoinImplementationAbstract.sol]: We will probably be moving all of the join adapters to dss, so this is fine.

Comment (gbalabasquer) [reply src/dss/GemJoinImplementationAbstract.sol]: What was the reason to move all the join adapters to `dss`?
I'm not really sure is the best idea to overload `dss` with a bunch of different adapters. We might have several types in the future.

**Pull request merged**